packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -40,6 +61,167 @@ abstract class BindingBase {
     assert(() { _debugInitialized = true; return true; });
   }
 
+  /// Called, in checked mode, to register service extensions.
```

Comment:
Ok I've made it so that service extensions are always registered, for now. Once we have actual profile builds we can revisit how this works.